### PR TITLE
[SHELL32][SHELL32_APITEST] Implement SHCreateFileDataObject

### PR DESCRIPTION
**Pull request by learn-more**

### CODE REVIEW

File: modules/rostests/apitests/shell32/SHCreateFileDataObject.cpp [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+        ok_ptr(out.ptd, NULL);
+        ok_int(out.dwAspect, (int)in.dwAspect);
+        ok_int(out.lindex, in.lindex);
+        ok_int(out.tymed, (int)in.tymed);
+        trace("in copied to out\n");
+    }
+}
+
+
+#define ok_wstri(x, y) \
```
Comment (HBelusca) [inline modules/rostests/apitests/shell32/SHCreateFileDataObject.cpp:68]: This macro seems to be redefined many times in this apitest/shell32... maybe consider putting it in a global header of this apitest?

File: dll/win32/shell32/CIDLDataObj.cpp [hunk 2]
```diff
@@ -400,3 +400,17 @@ HRESULT WINAPI SHCreateDataObject(PCIDLIST_ABSOLUTE pidlFolder, UINT cidl, PCUIT
...
+ *
+ */
+
+HRESULT WINAPI SHCreateFileDataObject(PCIDLIST_ABSOLUTE pidlFolder, UINT cidl, PCUITEMID_CHILD_ARRAY apidl, IDataObject* pDataInner, IDataObject** ppDataObj)
+{
+    if (pDataInner)
+        UNIMPLEMENTED;
+    return IDataObject_Constructor(NULL, pidlFolder, apidl, cidl, TRUE, ppDataObj);
+
+    return E_FAIL;
```
Comment (SergeGautherie) [inline dll/win32/shell32/CIDLDataObj.cpp:415]: Unreachable...

File: modules/rostests/apitests/shell32/SHCreateFileDataObject.cpp (hunk 1 above)
Comment (learn-more) [reply modules/rostests/apitests/shell32/SHCreateFileDataObject.cpp:68]: Feel free to, after this PR is merged.

**Comment by HBelusca** [comment]
As it appears this is not a documented function, I would suggest adding in the commit log that the details come from
http://undoc.airesoft.co.uk/shell32.dll/SHCreateFileDataObject.php

**Review by yagoulas — APPROVED** [review]

**Pull request merged**